### DISCUSSION TITLE
JavaScript - Dúvida com THIS dentro de uma FUNÇÃO em um OBJETO

### DISCUSSION THREAD

**Discussion by MarcosColucci** [frontendbr/forum#1963] [Perguntas e Respostas] [answered]
Olá, 

Estou iniciando com programação (nivel 0,1% no momento) e escolhi o JavaScript como ponto de partida. Acompanho um curso de javascript no Udemy, porém o retorno para questionamentos está muito moroso. 

A ideia é dentro deste objeto com dados de uma pessoa, criar uma função que automaticamente calcule a idade. Com o resultado, a função deve criar uma nova propriedade chamada _age_ com a idade. Porém, ao visualizar o log, recebo somente o resultado _undefined_.

Podem me ajudar a solucionar? 

const personData = {
    firstname: 'Marcos',
    lastname: 'Modesto',
    job: 'Constructor',
    birthyear: 1987,
    friends: ['Luiz', 'Fernando', 'Flavia'],
    hasDriversLicence: true,

    calcAge: function () {
        this.age = 2020 - this.birthyear;
        return this.age;
    }
};

console.log(personData.age);

**Comment by MarcosColucci** [comment, marked as answer]
consegui arrumar sozinho... valeu!

**Reply by robsonware** [reply]
coloca a solução aí pra galera po :p

**Comment by portothree** [comment]
A função `calcAge` precisa ser chamada para que o seu corpo seja executado.

Se chamar `personData.calcAge()` antes do `console.log` vai ter o resultado esperado.

**Comment by robsonware** [comment]
Usa solução seria:

```javascript
const personData = {
  firstname: 'Marcos',
  lastname: 'Modesto',
  job: 'Constructor',
  birthyear: 1987,
  friends: ['Luiz', 'Fernando', 'Flavia'],
  hasDriversLicence: true,
  age: function () { return new Date().getFullYear() - this.birthyear }
};
```
